packages/react-accordion/src/components/Accordion/useAccordionContext.ts

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import { createDescendantContext, useDescendant, useDescendantsInit } from '@reach/descendants';
+import { AccordionContext, AccordionDescendant, AccordionOpen, AccordionState } from './Accordion.types';
+import { useConst } from '@fluentui/react-utilities';
+
+export const accordionDescendantContext = createDescendantContext<AccordionDescendant>('AccordionDescendantContext');
+
+export const accordionContext = React.createContext<AccordionContext>(undefined!);
+
+export function useCreateAccordionContext(state: AccordionState) {
+  const { open, multiple, collapsible, onToggle } = state;
+  const isControlled = useConst(typeof open !== 'undefined');
+  const [descendants, setDescendants] = useDescendantsInit<AccordionDescendant>();
+  const [openItems, setOpenItems] = React.useState<AccordionOpen>(() => initializeOpenItems(state));
+
+  const requestToggle = React.useCallback(
+    (index: number) => {
+      onToggle?.(index);
+      if (!isControlled) {
+        setOpenItems(previousOpenItems =>
+          updateOpenItems(index, previousOpenItems, {
+            collapsible,
+            multiple,
+          }),
+        );
+      }
+    },
+    [collapsible, isControlled, multiple, onToggle],
+  );
```

Comment:
```suggestion
  const requestToggle = useEventCallback(
    (index: number) => {
      onToggle?.(index);
      if (!isControlled) {
        setOpenItems(previousOpenItems =>
          updateOpenItems(index, previousOpenItems, {
            collapsible,
            multiple,
          }),
        );
      }
    },
  );
```

Let's use `useEventCallback` (from `react-utilities`) there as `onToggle` can't be unstable, can be a closure, for example:

```tsx
<Accordion onToogle={() => {}} />
```

This will cause often changes of React's context and additional re-renders.

Reply:
IMHO it's the users problem to memoize the callbacks provided. honestly, it might be their intention to rerender on callback change, this might be a problem in some specific cases. @layershifter